src/main/java/usecases/CourseEvaluator/EvaluatorInteractor.java

```diff
@@ -0,0 +1,77 @@
+package usecases.CourseEvaluator;
+
+
+import entities.Course;
+import entities.CourseFactory;
+
+public class EvaluatorInteractor implements EvaluatorInputBoundary {
+   private EvaluatorOutputBoundary presenter;
+   private StudentDataAccess studentDataAccess;
+
+   private CourseDataAccess courseDbGateway;
+
+   private EvaluatorSessionDataAccess sessionDataAccess;
+
+   private CourseFactory courseFactory;
+
+   private CheckAnswer grade;
+
+
+   public EvaluatorInteractor(EvaluatorOutputBoundary presenter, CourseDataAccess courseDbGateway,
+                              EvaluatorSessionDataAccess sessionDataAccess, CourseFactory courseFactory,
+                              StudentDataAccess studentDataAccess, CheckAnswer grade) {
+        this.presenter = presenter;
+        this.courseDbGateway = courseDbGateway;
+        this.sessionDataAccess = sessionDataAccess;
+        this.courseFactory = courseFactory;
+        this.studentDataAccess = studentDataAccess;
+        this.grade = grade;
+    }
+
+    /**
+     * Use case for checking if a course exists, saving and evaluating  a students grade in percentages. If the course
+     * does not exist, an error is thrown, that informs students that the course does not exist.
+     * if course exists in database
+     * the student grade is saved to the Student data access.
+     * Then the presenter prepares success/fail views depending on each case.
+     *
+     * @param requestModel data structure packaging the courseId.
+     * @return String prepare success and fail view.
+     */
+
+
+    public String evaluate(EvaluatorRequestModel requestModel) {
+
+        if (!sessionDataAccess.courseExists(requestModel.getCourseId())) {
+            return presenter.prepareFailView();
+        }
```

Comment:
I'm not sure if we need this check.

At least the way I'm thinking about it is, if we get to your screen, we can successfully assume that Henry has saved the course to the session database so the course will always exist. 

Could you let me know if I'm thinking about it wrong or if you think we no longer require this?